src/js/live-tracker.js

```diff
@@ -44,67 +71,45 @@ class LiveTracker extends Component {
     }
   }
 
-  isBehind_() {
-    // don't report that we are behind until a timeupdate has been seen
-    if (!this.timeupdateSeen_) {
-      return false;
-    }
-    const liveCurrentTime = this.liveCurrentTime();
-    const currentTime = this.player_.currentTime();
-
-    // the live edge window is the amount of seconds away from live
-    // that a player can be, but still be considered live.
-    // we add 0.07 because the live tracking happens every 30ms
-    // and we want some wiggle room for short segment live playback
-    const liveEdgeWindow = (this.seekableIncrement_ * 2) + 0.07;
-
-    // on Android liveCurrentTime can bee Infinity, because seekableEnd
-    // can be Infinity, so we handle that case.
-    return liveCurrentTime !== Infinity && (liveCurrentTime - liveEdgeWindow) >= currentTime;
-  }
-
-  // all the functionality for tracking when seek end changes
-  // and for tracking how far past seek end we should be
+  /**
+   * all the functionality for tracking when seek end changes
+   * and for tracking how far past seek end we should be
+   */
   trackLive_() {
-    this.pastSeekEnd_ = this.pastSeekEnd_;
     const seekable = this.player_.seekable();
 
     // skip undefined seekable
     if (!seekable || !seekable.length) {
       return;
     }
 
-    const newSeekEnd = this.seekableEnd();
-
-    // we can only tell if we are behind live, when seekable changes
-    // once we detect that seekable has changed we check the new seek
-    // end against current time, with a fudge value of half a second.
-    if (newSeekEnd !== this.lastSeekEnd_) {
-      if (this.lastSeekEnd_) {
-        // we try to get the best fit value for the seeking increment
-        // variable from the last 12 values.
-        this.seekableIncrementList_ = this.seekableIncrementList_.slice(-11);
-        this.seekableIncrementList_.push(Math.abs(newSeekEnd - this.lastSeekEnd_));
-        if (this.seekableIncrementList_.length > 3) {
-          this.seekableIncrement_ = median(this.seekableIncrementList_);
-        }
-      }
+    const newTime = window.performance.now().toFixed(4);
+    const deltaTime = !this.lastTime_ ? 0 : (newTime - this.lastTime_) / 1000;
 
-      this.pastSeekEnd_ = 0;
-      this.lastSeekEnd_ = newSeekEnd;
-      this.trigger('seekableendchange');
-    }
+    this.lastTime_ = newTime;
 
-    // we should reset pastSeekEnd when the value
-    // is much higher than seeking increment.
-    if (this.pastSeekEnd() > this.seekableIncrement_ * 1.5) {
-      this.pastSeekEnd_ = 0;
-    } else {
-      this.pastSeekEnd_ = this.pastSeekEnd() + 0.03;
+    this.pastSeekEnd_ = this.pastSeekEnd() + deltaTime;
+
+    const liveCurrentTime = this.liveCurrentTime();
+    const currentTime = this.player_.currentTime();
+
+    // we are behind live if any are true
+    // 1. the player is paused
+    // 2. the user seeked to a location 2 seconds away from live
```

Comment:
This is something that may be worth reading in from VHS when available and assume we aren't live if we are a segment/target duration behind.

Reply:
Definitely nothing to do right now.